snuba/cli/perf.py

```diff
@@ -60,8 +59,14 @@ def perf(
     setup_logging(log_level)
 
     dataset = get_dataset(dataset_name)
-    if not local_dataset_mode():
-        logger.error("The perf tool is only intended for local dataset environment.")
+
+    if not all(
```

Comment:
You should not need to create a list to pass to the `all` function:

```
if not all(
   (storage.get_cluster().is_single_node() for storage in dataset.get_all_storages())
)
```
should work